### ISSUE TITLE
Module logger

### ISSUE THREAD

**Issue by JosePizarro3** [FAIRmat-NFDI/nomad-simulations#141]
@ladinesa @ndaelman-hu I have a question regarding the logger. Right now, I was passing it as an input of our functions, but I realize this might not be the best practice.

Do you know what is the best way of handling this? Define it in the class or in the modules?

**Comment by ladinesa** [comment]
the parse function takes in the logger as arg which should be passed on to other functions. i usually make it as class attrb. you can also define it as var in the module  in some cases

**Comment by ndaelman-hu** [comment]
The only solution is to share the same logger object among functions. That would be either a decorator or a class attribute, as Alvin suggested.
Given our structure, I would use a class decorator that prepends the section name.

**Comment by ndaelman-hu** [comment]
Let me first merge my branch before overhauling the schema at this scale, though.

**Comment by JosePizarro3** [comment]
Ok, I think class attribute is the most consistent one. I will fix it elsewhere later.

**Comment by ndaelman-hu** [comment]
> Ok, I think class attribute is the most consistent one. I will fix it elsewhere later.

Indeed, and if you apply it as a class decorator, your implementation will be more modular. How you prefer.

**Comment by JosePizarro3** [comment]
> > Ok, I think class attribute is the most consistent one. I will fix it elsewhere later.
> 
> Indeed, and if you apply it as a class decorator, your implementation will be more modular. How you prefer.

Why a class decorator? I think an input to each class `__init__` should be enough, right?

**Comment by ndaelman-hu** [comment]
> > > Ok, I think class attribute is the most consistent one. I will fix it elsewhere later.
> > 
> > 
> > Indeed, and if you apply it as a class decorator, your implementation will be more modular. How you prefer.
> 
> Why a class decorator? I think an input to each class `__init__` should be enough, right?

Please don't move it to the `__init__` args, that just displaces the issue, rather than solving it.
Instead, you want a function called during (class?) instantiation that reads in the context.
That functions could either be inherited or set as a class decorator to the sections. Either are fine for our usage.

**Comment by JFRudzinski** [comment]
@ladinesa @ndaelman-hu Is this issue or any part of this issue still relevant? If so, please clarify

**Comment by ladinesa** [comment]
In https://github.com/FAIRmat-NFDI/nomad-parser-plugins-simulation/blob/develop/src/nomad_simulation_parsers/parsers/utils/general.py#L104 I introduced a log decorator which we can adopt here.

**Comment by JFRudzinski** [comment]
replaced by #217
